src/TwigComponent/doc/index.rst

```diff
@@ -1058,6 +1058,63 @@ Exclude specific attributes:
       My Component!
     </div>
 
+Nested Attributes
+~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.17
+
+    The Nested Attributes feature was added in TwigComponents 2.17.
+
+You can have attributes that aren't meant to be used on the *root* element
+but one of its *descendants*. This is useful for, say, a dialog component where
+you want to allow customizing the attributes of the dialog's content, title,
+and footer. Here's an example of this:
+
+.. code-block:: html+twig
+
+    {# templates/components/Dialog.html.twig #}
+    <div{{ attributes }}>
+        <div{{ attributes.nested('title') }}>
+            {% block title %}Default Title{% endblock %}
+        </div>
+        <div{{ attributes.nested('body') }}>
+            {% block content %}{% endblock %}
+        </div>
+        <div{{ attributes.nested('footer') }}>
+            {% block footer %}Default Footer{% endblock %}
+        </div>
+    </div>
+
+    {# render #}
+    <twig:Dialog class="foo" title:class="bar" body:class="baz" footer:class="qux">
+        Some content
+    </twig:MyDialog>
+
+    {# output #}
+    <div class="foo">
+        <div class="bar">
+            Default Title
+        </div>
+        <div class="baz">
+            Some content
+        </div>
+        <div class="qux">
+            Default Footer
+        </div>
+    </div>
+
+The nesting is recursive so you could potentially do something like this:
```

Comment:
I'm on very small details but this is not really recursive and more a tree here (in the meaning a parent call child, but never call itself back)

This paragraph and your following example make me want to do two small TwigComponent demos 
* one litterally recursive where you pass to the child a level max and something a bit dynamic (oh i think i have n idea, poke me if your'e around) and the recursion goes until the level is reached
* a form demo with a cool theme, just like you did.

WDYT ? 


(and we need to start the TwigComponent demo)